moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -642,7 +603,7 @@ moveit::core::Transforms& PlanningScene::getTransformsNonConst()
   {
     // The only case when there are no transforms is if this planning scene has a parent. When a non-const version of
     // the planning scene is requested, a copy of the parent's transforms is forced
-    scene_transforms_.reset(new SceneTransforms(this));
+    scene_transforms_ = std::make_shared<moveit::core::Transforms>(getRobotModel()->getModelFrame());
     scene_transforms_->setAllTransforms(parent_->getTransforms().getAllTransforms());
```

Comment:
Not sure, why we copy all parent's transforms here. This defeats the concept of diffs.

Reply:
The `PlanningScene` has a rather weak definition of "diff" for some interfaces, where `diff` means "use the parent's data structure unless you defined your own (in which case the parent is not looked at at all). For a full diff, you would have to maintain deletions from the parent's structure as well and this is definitely not implemented everywhere.
I would expect it's not implemented for the transforms.